md_client/client.py

```diff
@@ -36,17 +42,36 @@ async def sync_with_remote(self):
                 action = await self.reader.read()
                 self.db_actions.handle_protobuf(action)
 
-    def pull_db(self):
-        pass
+    async def send_db_info(self):
+        message = md_pb2.DBInfo(db_name=self.db_name, client_id=self.client_id)
+        self.writer.write(message.SerializeToString())
 
-    def send_protobuf(self, protobuf):
-        logging.debug(f"Sent protobuf {protobuf}")
-        self.writer.write(protobuf)
+    async def pull_db(self):
+        db_hash = get_db_md5(self.db_name)
+        message = md_pb2.GetDBHash(db_hash=db_hash)
+        self.writer.write(message.SerializeToString())
 
-    def add_item(self, item):
-        pass
-        # message = self.md_protocol.create_message(ADD, paten)
-        # self.send_protobuf(message)
+        db_state = md_pb2.GetDBState()
+        db_state.ParseFromString(await self.reader.read())
 
-    def delete_item(self, item):
-        pass
+        if db_state.state == md_pb2.DBState.NOT_SYNCED:
+            logging.info("DBs are not synced! Syncing dbs, this might overwrite data")
+            db_data = md_pb2.GetDB()
+            db_data.ParseFromString(await self.reader.read())
+            with open(self.db_name, 'wb') as f:
+                f.write(db_data.db_file)
+
+
+def send_protobuf(self, protobuf):
+    logging.debug(f"Sent protobuf {protobuf}")
+    self.writer.write(protobuf)
```

Comment:
Maybe we should call async writer.drain after write? When I tried without drain it caused me problems